src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/sqlinternaltransaction.cs

```diff
@@ -591,20 +590,16 @@ internal void Zombie()
 
         private void ZombieParent()
         {
-            if (null != _parent)
+            if (_parent != null && _parent.TryGetTarget(out SqlTransaction parent))
```

Comment:
As per current design, should we ensure `_parent` is set to `null` even if `_parent.TryGetTarget` returns false?

Reply:
if parent is null TryGetTarget will return false, I checked the code in runtime. The corollary is also true, if TryGetTarget returns true then parent is guaranteed not to be null.

Reply:
I meant reverse, if **parent** is **not** null, will **TryGetTarget** always return **true**?
If no, then we have a case where parent will not be zombied to null, if there was no target attached.

Reply:
> if parent is not null, will TryGetTarget always return true?
yes.

Reply:
```csharp
        public bool TryGetTarget([MaybeNullWhen(false), NotNullWhen(true)] out T target)
        {
            // Call the worker method that has more performant but less user friendly signature.
            T? o = this.Target;
            target = o!;
            return o != null;
        }
```

https://github.com/dotnet/runtime/blob/efd818b90d61a81b73b20f3d1720279f993faa2c/src/libraries/System.Private.CoreLib/src/System/WeakReference.T.cs#L54-L60

You can tell from the naming that is is old, o wouldn't pass api review today.

Reply:
Well, I'd say it depends on whether there was an item added to it, but is not going to return "true" _by design_. Just to keep things consistent with current design, I'd recommend putting it in second block and setting parent to _null_ never-the-less.

Reply:
Done.